src/hooks/useFetchWeather.js

```diff
@@ -0,0 +1,109 @@
+import { useState, useEffect } from "react";
+import { mapCurrentWeatherData, mapForecastData } from "../lib/mappers";
+import { WEATHER_TYPES, API, UNITS } from "../constants";
+import { API_KEY } from "../config";
+
+const useFetchWeather = (initialLocation) => {
+  const [currentWeatherData, setCurrentWeatherData] = useState({});
```

Comment:
Prefer to use null instead of empty object for those cases (when working with TS you're going to thank me)
```suggestion
  const [currentWeatherData, setCurrentWeatherData] = useState(null);
```

Reply:
Only with empty objects? Is it okay with empty arrays?